### ISSUE TITLE
[HOLD for payment 2023-07-17] [$1000] IOS - Icon is not set when https://new.expensify.com is added to home screen on iOS devices

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21064]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to https://new.expensify.com/ in iOS Safari
2. Tap on share icon from toolbar
3. Select Add to Home Screen option
4. Tap Add on the following screen
5. Observe the icon that gets added to Home Screen

## Expected Result:
Home screen should show New Expensify icon

## Actual Result:
A generic icon with letter 'N' is shown

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.29.2

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/3aa1f7be-5938-4a9d-9197-e52657ce6295

https://github.com/Expensify/App/assets/93399543/c4d86d2d-6e85-4262-9e5e-dcf8fe5a9edd

**Expensify/Expensify Issue URL:**

**Issue reported by:** @aswin-s

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686538637077189

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01221dcc554041530c</li>
        <li>Upwork Job ID: 1673346830813540352</li>
        <li>Last Price Increase: 2023-06-26</li>
    </ul>
</details>

**Comment by eh2077** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Icon is not set when https://new.expensify.com is added to home screen on iOS devices.

### What is the root cause of that problem?

The root cause of this issue is that in file [index.html](https://github.com/Expensify/App/blob/main/web/index.html), we didn't set `apple-touch-icon` meta tag there.

### What changes do you think we should make in order to solve the problem?

To fix this issue, we add following line before this [line](https://github.com/Expensify/App/blob/bee8fd3e1634fa11761435ef405a3d3efd654b07/web/index.html#L126) of web/index.html

```html
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
```

We'll also need to ask the design team to design an icon for it.

### What alternative solutions did you explore? (Optional)

N/A

**Comment by adelekennedy** [comment]
ugh - this got assigned when I was already ooo, reassigning

**Comment by puneetlath** [comment]
Seems like an easy thing to address. Adding the external label.

**Comment by mollfpr** [comment]
The [proposal](https://github.com/Expensify/App/issues/21064#issuecomment-1597561421) from @eh2077 looks good to me, and yes we do need the icon.

🎀 👀 🎀 C+ reviewed!

**Comment by puneetlath** [comment]
@marcochavezf what do you think about the proposal? 

Also, adding the design label for help with the icon.

**Comment by shawnborton** [comment]
Yup, happy to help here. I think this asset should work: 
![apple-touch-icon](https://github.com/Expensify/App/assets/2319350/a166f454-c063-4757-875d-4cf80cf43afe)

[apple-touch-icon.png.zip](https://github.com/Expensify/App/files/11894916/apple-touch-icon.png.zip)

**Comment by eh2077** [comment]
@shawnborton Thanks for providing the icon asset. Yes, it looks perfect, see below demo video

https://github.com/Expensify/App/assets/117511920/c1bf81f5-c6f5-49cb-a2a4-46df616dc059

**Comment by shawnborton** [comment]
Great!

Then, curious where the icon here is coming from?
<img width="301" alt="image" src="https://github.com/Expensify/App/assets/2319350/01f9e6af-01a6-4f7a-8f20-41f2100e1775">

**Comment by eh2077** [comment]
> Great!
> 
> Then, curious where the icon here is coming from? <img alt="image" width="301" src="https://user-images.githubusercontent.com/2319350/249495268-01f9e6af-01a6-4f7a-8f20-41f2100e1775.png">

It looks like the [favicon.png](https://github.com/Expensify/App/blob/main/web/favicon.png) we're using in Browser tab.

**Comment by eh2077** [comment]
@mollfpr We also need to include the icon in [webpack](https://github.com/Expensify/App/blob/94e265653f83505e581e99ac0ec0d3ea30e23998/config/webpack/webpack.common.js#L84) file you're going to double check it.

**Comment by marcochavezf** [comment]
The proposal LGTM, assigning @eh2077 🚀

**Comment by eh2077** [comment]
@marcochavezf We're blocked on testing the PR on Android mobile Chrome. My emulator env can't duplicate what @mollfpr encountered in their env. It's weird that adding the meta tag for iOS impacts the behaviour on Android mobile Chrome. Can you help us move it forward? 

I investigated the issue and found https://github.com/Expensify/App/pull/21856#issuecomment-1615516319 and https://github.com/Expensify/App/pull/21856#issuecomment-1617228769

> I guess it's related to the legacy usage of `rel="shortcut icon"`.
> It looks like `rel="shortcut icon"` is not recommended for modern browsers and it's recommended to use rel="icon", see [rel="shortcut icon" considered harmful](https://mathiasbynens.be/notes/rel-shortcut-icon) and [How to Favicon in 2023: Six files that fit most needs](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) FYI.

**Comment by puneetlath** [comment]
@eh2077 are you looking for specific feedback from Marco? What exactly do you need him to do?

**Comment by eh2077** [comment]
@puneetlath Yes, we got some trouble in testing the PR but it's solved later. Now the PR has been merged into main and we just need to wait for the automation.

**Comment by puneetlath** [comment]
Ah, great!

**Comment by puneetlath** [comment]
Sent everyone Upwork offers.

**Comment by mollfpr** [comment]
>[@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
 [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

No offending PR.

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

This is going to be one-time bug, so the regression step should be enough.

> [@mollfpr] Determine if we should create a regression test for this bug.
 [@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


**Mobile Web Safari**
1. Navigate to https://staging.new.expensify.com/ in iOS Safari
2. Tap on the share icon from the toolbar
3. Select **_Add to Home Screen_** option
4. Tap Add on the following screen
5. **Verify** the Expensify icon that gets added to Home Screen
6. Click the icon bookmark just added from Home Screen
7. **Verify** that it opens https://new.expensify.com/

**Mobile Web Chrome**
1. Navigate to https://staging.new.expensify.com/ in Android Chrome
2. Tap on three dot icon in the top right corner
3. Select **_Add to Home Screen_** option
4. Tap Add on the following screen
5. **Verify** the Expensify icon that gets added to Home Screen
6. Click the icon bookmark just added from Home Screen
7. **Verify** that it opens https://new.expensify.com/

**Comment by puneetlath** [comment]
Sorry for the delay here. All paid!
